lib/repositories/explore/explore_repository.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:Medito/constants/constants.dart';
+import 'package:Medito/models/models.dart';
+import 'package:Medito/services/network/dio_api_service.dart';
+import 'package:Medito/services/network/dio_client_provider.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'explore_repository.g.dart';
+
+abstract class ExploreRepository {
+  Future<ExploreModel> fetchExploreResult(String query);
+}
+
+class ExploreRepositoryImpl extends ExploreRepository {
+  final DioApiService client;
+
+  ExploreRepositoryImpl({required this.client});
+
+  @override
+  Future<ExploreModel> fetchExploreResult(String query) async {
+    var res = await client
```

Comment:
res! can avoid using abbreviations?

Reply:
We are writing so that people who read it in the future can understand what's going on without having to think too much. It's good to keep this in mind when typing :)

Reply:
Ah, sorry my bad I only changed the name but didn't look at the abbreviation.